extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapGestures.kt

```diff
@@ -0,0 +1,121 @@
+package com.mapbox.maps.extension.androidauto
+
+import androidx.car.app.SurfaceCallback
+import com.mapbox.common.Logger
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.ScreenCoordinate
+import com.mapbox.maps.plugin.animation.camera
+
+/**
+ * This class captures gesture events from Android Auto's SurfaceCallback, and then applies them
+ * to the Mapbox map camera. If you would like to implement or customize the map gestures provided,
+ * this class is open to override. You can set your own gestures with [MapboxCarMap.setGestures].
+ */
+@OptIn(MapboxExperimental::class)
+open class MapboxCarMapGestures {
+
+  /**
+   * Allows you to implement or observe the map scroll gesture handler. The surface is
```

Comment:
How's different `implement` vs `observe` for the user?

Reply:
The `CarCameraController` added to the example `observes` the gesture behavior.

``` kotlin
class CarCameraController : MapboxCarMapGestures(), MapboxCarMapObserver {
    override fun onScroll(
..
  ) {
    isTrackingPuck = false
    super.onScroll(mapboxCarMapSurface, visibleCenter, distanceX, distanceY)
  }
}
```

If you want to `implement` the gesture behavior. You don't have to call `super`, and you create your own version.